File: app/src/main/java/com/itus/u_money/mvp/model/dao/BillDAO.java
package com.itus.u_money.mvp.model.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.itus.u_money.mvp.model.Bill;

import java.util.List;

@Dao
public interface BillDAO {
    @Query("SELECT * FROM `Bill`")
    List<Bill> getAll();

    @Query("SELECT * FROM `Bill` WHERE id IN (:billIds)")
    List<Bill> loadAllByIds(int[] billIds);

    @Insert
    void insertAll(Bill... bills);

    @Update
    void updateAll(Bill... bills);

    @Delete
    void deleteAll(Bill... bills);
}

File: app/src/main/java/com/itus/u_money/model/dao/BillDAO.kt
package com.itus.u_money.model.dao

import androidx.room.*
import com.itus.u_money.model.Bill

@Dao
interface BillDAO {
    @Query("SELECT * FROM `Bill`")
     fun getAll(): List<Bill>?

    @Query("SELECT * FROM `Bill` WHERE id IN (:billIds)")
    fun loadAllByIds(billIds: IntArray?): List<Bill?>?

    @Insert
    fun insertAll(vararg bills: Bill?)

    @Update
    fun updateAll(vararg bills: Bill?)

    @Delete
    fun deleteAll(vararg bills: Bill?)
}